chia/full_node/mempool_check_conditions.py

```diff
@@ -445,26 +148,26 @@ def mempool_check_conditions_dict(
         cvp: ConditionWithArgs
         for cvp in con_list:
             error: Optional[Err] = None
-            if cvp.opcode is ConditionOpcode.ASSERT_MY_COIN_ID:
-                error = mempool_assert_my_coin_id(cvp, unspent)
-            elif cvp.opcode is ConditionOpcode.ASSERT_COIN_ANNOUNCEMENT:
-                error = mempool_assert_announcement(cvp, coin_announcement_names)
-            elif cvp.opcode is ConditionOpcode.ASSERT_PUZZLE_ANNOUNCEMENT:
-                error = mempool_assert_announcement(cvp, puzzle_announcement_names)
-            elif cvp.opcode is ConditionOpcode.ASSERT_HEIGHT_ABSOLUTE:
+            if cvp.opcode is ConditionOpcode.ASSERT_HEIGHT_ABSOLUTE:
                 error = mempool_assert_absolute_block_height_exceeds(cvp, prev_transaction_block_height)
             elif cvp.opcode is ConditionOpcode.ASSERT_HEIGHT_RELATIVE:
                 error = mempool_assert_relative_block_height_exceeds(cvp, unspent, prev_transaction_block_height)
             elif cvp.opcode is ConditionOpcode.ASSERT_SECONDS_ABSOLUTE:
                 error = mempool_assert_absolute_time_exceeds(cvp, timestamp)
             elif cvp.opcode is ConditionOpcode.ASSERT_SECONDS_RELATIVE:
                 error = mempool_assert_relative_time_exceeds(cvp, unspent, timestamp)
+            elif cvp.opcode is ConditionOpcode.ASSERT_MY_COIN_ID:
+                assert False
```

Comment:
It's recommend to not use `assert` for program flow, since `python -O` removes asserts from code. Consider `raise SomeExceptionType` instead.

Reply:
On the other hand, maybe an assert is exactly what you want here?

Reply:
yeah, these conditions aren't supposed to show up here anymore. so, I would consider it a programmer error if they do